storage/src/main/java/org/apache/kafka/storage/internals/log/RemoteIndexCache.java

```diff
@@ -421,6 +425,7 @@ public void close() {
                 // Initiate shutdown for cleaning thread
                 boolean shutdownRequired = cleanerThread.initiateShutdown();
                 // Close all the opened indexes to force unload mmap memory. This does not delete the index files from disk.
+                internalCache.cleanUp();
```

Comment:
@divijvaidya 

I don't understand the below comment. Could you please explain? Is it OK to do cleanup before closing the cache otherwise the newly added tests are failing? (testRemoveItem and testRemoveNonExistentItem) Thanks!

> // Note that internal cache does not require explicit cleaning/closing. We don't want to invalidate or cleanup
                // the cache as both would lead to triggering of removal listener.

Reply:
see: https://github.com/apache/kafka/pull/13850#discussion_r1229087123

Think of this cache as a hashmap. When we are closing the cache, garbage collector will automatically reclaim memory. We do not have to explicitly release any pinned resources. That is why calling cleanUp is not required. Also, cleanUp actually performs maintenance activity such as cache invalidation etc. which we don't really want. 

Let me take a look at tests to see why they are failing.

Reply:
Fixed this issue with the latest commit.